Header header
string location
bool in_control